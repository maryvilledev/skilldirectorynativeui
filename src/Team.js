import React, { Component } from 'react'
import { Text, View, ScrollView, Button, RefreshControl, Alert } from 'react-native'
import axios from 'axios'
import { API_URL } from './Env'
import { StackNavigator } from 'react-navigation'
import AddForm, {FORM_TYPE} from './AddForm'

import { centerLayout, scrollLayout, textStyles } from './Styles'

const api = API_URL;

class Team extends Component {
  static navigationOptions = {
    title: "Team Members",
    tabBar: {
      label: "Team"
    }
  }
  constructor() {
    super();
    this.state = {
      modalVisible: false,
      refreshing: true,
      teamMembers: [],
    };

    this.addTeamMember = this.addTeamMember.bind(this)
    this.deleteTeamMember = this.deleteTeamMember.bind(this)
    this.doRefresh = this.doRefresh.bind(this)
  }
  componentDidMount() {
    this.doRefresh();
  }

  doRefresh() {
    this.setState({refreshing: true});
    axios.get(`${api}/teammembers/`)
      .then((result) => {
        const teamMembers = result.data.slice();
        this.setState({
          teamMembers: teamMembers,
          refreshing: false,
        });
      })
      .catch((err) => {
        console.log(`Error fetching team members: ${err}`);
      });
  }
  addTeamMember(name, title) {
    axios.post(`${api}/teammembers/`, {name: name, title: title})
      .then(() => {
        this.doRefresh()
      })
      .catch(err => {
        console.error(err)
      })
  }
  deleteTeamMember(id){
    axios.delete(`${api}/teammembers/${id}`)
      .then(() => {
        this.doRefresh()
      })
      .catch(err => {
        console.error(err)
      })
  }
  render() {
    const { navigate } = this.props.navigation;
    const forms = [
      {type: FORM_TYPE.TEXT, label: "Name"},
      {type: FORM_TYPE.TEXT, label: "Title"},
    ];
    const members = this.state.teamMembers.map((member, index) => { return (
      <Button
        title={member.name}
        key={index}
        onPress={() => navigate('Detail', {member: member, deleteCallback: () => {
          this.deleteTeamMember(member.id);
        }})}
      />
    )});
    return (
      <View style={{flex: 1}}>
        <ScrollView
          style={scrollLayout}
          refreshControl = {
            <RefreshControl
              refreshing={this.state.refreshing}
              onRefresh={this.doRefresh}
            />
          }
        >
          {members}
        </ScrollView>
        <Button
          title="Add"
          onPress={() => this.setState({modalVisible: true})}
        />
        <AddForm
          forms={forms}
          visible={this.state.modalVisible}
          onClose={() => this.setState({modalVisible: false})}
          onSubmit={vals => this.addTeamMember(...vals)}
        />
      </View>
    )
  }
}

class Detail extends Component {
  static navigationOptions = {
    title: ({state}) => state.params.member.name,
    tabBar: {
      label: "Team"
    }
  }
  render() {
    const { goBack } = this.props.navigation;
    const member = this.props.navigation.state.params.member;
    const deleteCallback = this.props.navigation.state.params.deleteCallback;
    return (
      <View style={centerLayout}>
        <Text style={textStyles.large}>{member.title}</Text>
        <Button
          onPress={() => {Alert.alert('Delete Team Member', 'Are you sure?', [
            {text: 'Yes', onPress: () => {
              deleteCallback();
              goBack();
            }, style: 'destructive'},
            {text: 'No', style: 'cancel'},
          ])}}
          title="Delete"
          color="red"
        />
      </View>
    )
  }
}

const TeamNavigator = StackNavigator({
  All: {screen: Team},
  Detail: {screen: Detail}
});

export default TeamNavigator;
