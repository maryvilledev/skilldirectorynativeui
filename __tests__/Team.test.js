import 'react-native'
import React from 'react'
import { Team, Detail } from './../src/Team.js'

import renderer from 'react-test-renderer'

const mockNavigator = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  state: {
    params: {
      member: {
        name: 'Donnie Darko',
        title: 'Destructor'
      },
      deleteCallback: jest.fn()
    }
  }
}

describe('<Team />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Team navigation={mockNavigator}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('<Detail />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Detail navigation={mockNavigator}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
