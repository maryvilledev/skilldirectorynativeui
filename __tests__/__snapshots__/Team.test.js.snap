exports[`<Detail /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
      }
    }>
    Destructor
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "red",
            },
          ]
        }>
        Delete
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Team /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <ScrollView
    refreshControl={
      <RefreshControl
        onRefresh={[Function]}
        refreshing={true} />
    }
    style={
      Object {
        "alignSelf": "stretch",
      }
    } />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }>
        Add
      </Text>
    </View>
  </View>
  <View>
    <Modal
      animationType="slide"
      onRequestClose={[Function]}
      transparent={false}
      visible={false}>
      <View
        style={
          Object {
            "height": 20,
          }
        } />
      <ScrollView
        style={
          Object {
            "alignSelf": "stretch",
          }
        }>
        <View
          style={
            Object {
              "borderRadius": 3,
              "borderWidth": 0.5,
              "margin": 10,
              "marginVertical": 5,
              "overflow": "hidden",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
              }
            }>
            Name: 
          </Text>
          <TextInput
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "borderColor": "grey",
                "borderWidth": 1,
                "height": 40,
              }
            }
            value="" />
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
              "borderWidth": 0.5,
              "margin": 10,
              "marginVertical": 5,
              "overflow": "hidden",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
              }
            }>
            Title: 
          </Text>
          <TextInput
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "borderColor": "grey",
                "borderWidth": 1,
                "height": 40,
              }
            }
            value="" />
        </View>
      </ScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                Object {},
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }>
              Submit
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                Object {},
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }>
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
